src/yourdfpy/urdf.py

```diff
@@ -170,6 +170,7 @@ class Robot:
     name: str
     links: List[Link] = field(default_factory=list)
     joints: List[Joint] = field(default_factory=list)
+    materials: Dict[str, Material] = field(default_factory=dict)
```

Comment:
Could we make this a `List`, just like the `links` and `joints`?  I don't see those to be fundamentally different.

Reply:
The dictionary here gives us faster lookups in ``apply_visual_color``:
```python
    color = (
        materials[visual.material.name].color
        if visual.material.name
        else visual.material.color
    )
```
With a list we would get linear (instead of average constant time) lookups there.

Reply:
The change is really minimal, basically:
```
# in _create_maps()
self._material_map = {}
for m in self.robot.materials:
  self._material_map[m.name] = m
```
and
```
# in apply_visual_color(self, ...) or call it with apply_visual_color(materials=self._material_map)
color = (
        self._material_map[visual.material.name].color
        if visual.material.name
        else visual.material.color
    )
```

Reply:
Gotcha, I hadn't noticed that you already faced the same question with links and joints. As of f0e38c2c8d851301afef8a30d06529edb382bd20 materials are implemented the same way, including the equality comparison which was missing.

Reply:
Beautiful, thanks!